python_dwd/file_path_handling/file_index_creation.py

```diff
@@ -0,0 +1,84 @@
+""" file index creation for available DWD station data """
+from pathlib import PurePosixPath
+import re
+from functools import lru_cache
+import ftplib
+import pandas as pd
+
+from python_dwd.constants.access_credentials import DWD_PATH, DWD_SERVER
+from python_dwd.constants.metadata import ARCHIVE_FORMAT, STATID_REGEX
+from python_dwd.download.ftp_handling import FTP
+from python_dwd.enumerations.column_names_enumeration import DWDMetaColumns
+from python_dwd.enumerations.parameter_enumeration import Parameter
+from python_dwd.enumerations.period_type_enumeration import PeriodType
+from python_dwd.enumerations.time_resolution_enumeration import TimeResolution
+
+
+@lru_cache(maxsize=None)
```

Comment:
Why do you introduce a cache here?

Reply:
The function is the minimum requirement for the other main functions. As it is called within those functions e.g. **create_file_list_for_dwd_server** the cache can also be reset in those functions with an argument like **create_new_file_index**. Also by using a cache instead of a file based system, all kinds of logics can be ommited e.g. checking for an existing file index on hard drive, writing a file index to hard drive if non exists...

Reply:
Does this really work? My experiences with lru.cache are limited so seems legit

Reply:
Works like a charme.